compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -365,9 +380,20 @@ private void generateSplits(BIRFunction function, List<Split> possibleSplits,
             Split currSplit = possibleSplits.get(splitNum);
             splitNum += 1;
             BIRNonTerminator lastInstruction = basicBlocks.get(currSplit.endBBNum).instructions.get(currSplit.lastIns);
-            BIROperand newCurrentBBTerminatorLhsOp = new BIROperand(lastInstruction.lhsOp.variableDcl);
-            BIRFunction newBIRFunc = createNewBIRFunctionAcrossBB(function, newFuncName, currSplit, newBBNum,
-                    fromAttachedFunction);
+            BType newFuncReturnType = lastInstruction.lhsOp.variableDcl.type;
+            BIROperand splitLastInsLhsOp = new BIROperand(lastInstruction.lhsOp.variableDcl);
+            BIROperand splitFuncCallResultOp;
+            if (currSplit.returnValAssigned) {
+                splitFuncCallResultOp = generateTempLocalVariable(function, newFuncReturnType);
+                LinkedHashSet<BType> memberTypes = new LinkedHashSet<>(2);
+                memberTypes.add(newFuncReturnType);
+                memberTypes.add(symbolTable.errorType);
+                newFuncReturnType = new BUnionType(null, memberTypes, false, false);
```

Comment:
Shall we move this union type creation to a separate method?

Reply:
Will look into it.